src/excel-markdown-helpers.ts

```diff
@@ -0,0 +1,147 @@
+ */
+export function addAlignmentSyntax(markdownRows: string[], columnWidths: number[], colAlignments: string[]):string[] {
+    // Deepcopy: https://stackoverflow.com/questions/35504310/deep-copy-an-array-in-angular-2-typescript#35504348
+    let result = Object.assign([], markdownRows);
+    
+    // Insert the markdown alignment syntax as second row in the output table
+    result.splice(1, 0,
+        "|" + columnWidths.map(function (width, index) {
+            let {prefix, postfix, adjust} = calculateAlignmentMarkdownSyntaxMetadata(colAlignments[index]);
+            return prefix + Array(columnWidths[index] + 3 - adjust).join("-") + postfix;
+        }).join("|") + "|");
+    return result;
+}
+
+/**
+ * Derives the Markdown Alignment syntax metadata on how to align the cell text
+ * @param alignment The cell text alignment (l = left, c = center, r = right)
+ */
+export function calculateAlignmentMarkdownSyntaxMetadata(alignment: string) : TableCellAlignment {
+    
+    switch (alignment) {
+        case ALIGNED_LEFT: return ALIGNED_LEFT_SYNTAX
+        case ALIGNED_CENTER: return ALIGNED_CENTER_SYNTAX
+        case ALIGNED_RIGHT: return ALIGNED_RIGHT_SYNTAX
+        default: return ALIGNED_LEFT_SYNTAX
+    }
+}
+
+/**
+ * Derives alignments from excel table data and calculates
+ * column padding for each column
+ * @param rows Table of data from Excel
+ */
+export function getColumnWidthsAndAlignments(rows: string[][]) : TableSizeMetadata {
+    let colAlignments: string[]=[];
+    return {
+        columnWidths: rows[0].map(function (column, columnIndex) {
+            // Derive column alignment from excel column header text
+            let alignment = columnAlignment(column);
+            colAlignments.push(alignment);
+            
+            // Replace header text with unaligned header text
+            column = column.replace(COLUMN_ALIGNMENT_REGEX, "");
+            rows[0][columnIndex] = column;
+            
+            // Return max width to support all columns for the entire table
+            // Applied across all rows
+            // TODO: Refactor the above to separate steps to make the intent clearer
+            return columnWidth(rows, columnIndex);
+        }),
+        colAlignments: colAlignments
+    };
+}
+
+/**
+ * Maps the original header alignment to an internal alignment
+ * @param columnHeaderText The original Excel column header text
+ */
+export function columnAlignment(columnHeaderText: string): string {
+            
+    var m = columnHeaderText.match(COLUMN_ALIGNMENT_REGEX);
+
+    if (m) {
+        var alignChar = m[1][1].toLowerCase();
+        return columnAlignmentFromChar(alignChar);
+    }
+
+    // Default to left alignment
+    return ALIGNED_LEFT;
+}
+
+/**
+ * Maps the original column alignment to an internal alignment
+ * @param alignChar Alignment character (l = left, c = center, r = right)
+ */
+export function columnAlignmentFromChar(alignChar: string) {
```

Comment:
What's the purpose of the switch case? Seems a bit excessive.

Reply:
Possibly overkill for what it's being used for. I could just move it inline if you prefer

Reply:
Not sure if it's necessary to have it since markdown default is left alignment. So if somebody were to use an invalid ^ prefix it would simply default anyway. Do you agree?

Reply:
Yep. Agreed. Happy with the below?

```
export function columnAlignment(columnHeaderText: string): string {

    var m = columnHeaderText.match(COLUMN_ALIGNMENT_REGEX);

    if (m) {
        var alignChar = m[1][1].toLowerCase();
        //return columnAlignmentFromChar(alignChar);
        if (alignChar === ALIGNED_RIGHT) return ALIGNED_RIGHT;
        if (alignChar === ALIGNED_CENTER) return ALIGNED_CENTER;
    }

    // Default to left alignment
    return ALIGNED_LEFT;
}
```

Reply:
Happy.